sqlmesh/core/engine_adapter/base.py

```diff
@@ -863,30 +864,42 @@ def _drop_object(
         """
         self.execute(exp.Drop(this=exp.to_table(name), kind=kind, exists=exists, **drop_args))
 
-    def get_alter_expressions(
+    def get_alter_operations(
         self,
         current_table_name: TableName,
         target_table_name: TableName,
-    ) -> t.List[exp.Alter]:
+    ) -> AlterTableOperations:
```

Comment:
Since the alter operations for BigQuery cannot be expressed in SQL, I piggybacked `SourceQuery` to allow arbitrary side-effects like making API calls